src/plugins/telemetry/server/plugin.ts

```diff
@@ -157,6 +158,19 @@ export class TelemetryPlugin implements Plugin<TelemetryPluginSetup, TelemetryPl
       sendTo: this.initialConfig.sendUsageTo === 'prod' ? 'production' : 'staging',
     });
 
+    analytics.registerContextProvider({
+      name: 'telemetry labels',
+      context$: this.config$.pipe(map(({ labels }) => ({ labels }))),
+      schema: {
+        labels: {
+          type: 'pass_through',
+          _meta: {
+            description: 'Custom labels added to the telemetry.labels config in the kibana.yml',
+          },
```

Comment:
If we choose to go with a strict validation, we can be explicit here (although I don't think we need to because the keys are internal)
```suggestion
          properties: {
            branch: {
              type: 'keyword',
              _meta: { description: 'The GitHub branch', optional: true },
            },
            ciBuildJobId: {
              type: 'keyword',
              _meta: { description: 'The CI Job ID', optional: true },
            },
            ciBuildId: {
              type: 'keyword',
              _meta: { description: 'The CI Build ID', optional: true },
            },
            ciBuildNumber: {
              type: 'long',
              _meta: { description: 'The CI Build Number', optional: true },
            },
            ftrConfig: {
              type: 'keyword',
              _meta: { description: 'The FTR config', optional: true },
            },
            git_rev: {
              type: 'keyword',
              _meta: { description: 'The current commit hash', optional: true },
            },
            isPr: {
              type: 'boolean',
              _meta: { description: 'Is a build triggered by a PR', optional: true },
            },
            prId: { type: 'keyword', _meta: { description: 'The PR ID', optional: true } },
            journeyName: {
              type: 'keyword',
              _meta: { description: 'The journey name in the performance tests', optional: true },
            },
            testBuildId: {
              type: 'keyword',
              _meta: { description: 'The Test Build ID', optional: true },
            },
            testJobId: {
              type: 'keyword',
              _meta: { description: 'The Test Job ID', optional: true },
            },
          }
```